Documentation/Manual/Texinfo/extensions.texi

```diff
@@ -321,3 +321,59 @@ image. Valid values are @code{:png}, @code{:jpg}, @code{:jpeg},
 An image describing the output to the @code{stream-var} stream will be
 returned as an rgba pattern (of class @code{clime:image-pattern}).
 @end defmac
+
+@section Updating frame instances after frame class redefinition
+
+Each application frame is an instance of a subclasses of
+@code{clim:application-frame} which is usually defined using
+@code{clim:define-application-frame}.  As such, the application frame
+will immediately reflect changes to slots, superclasses and specialized
+methods made by redefining its class.  In contrast, changes to panes,
+layouts and command tables which also occur when a frame class is
+redefined, for example as a result of changing and re-evaluating the
+@code{clim:define-application-frame} form, are not reflected in already
+existing instances by default.
+
+@mcclim provides an alternative behavior in which the latter aspects are
+also updated in already existing application frames.  This behavior is
+selected if the @code{(:update-instances-on-redefinition t)} option is
+supplied in the @code{clim:define-application-frame} form in
+question@footnote{Note that it is currently not possible to change this
+option after the initial definition of an application frame class.  This
+may be desirable anyway since changing this option after creating
+instances of the class would have unclear semantics.}.  When an
+application frame class which has been defined with this option is
+redefined by re-evaluating the @code{clim:define-application-frame} form
+or reinitializing the class in some other way, the following changes are
+applied to each existing instance of that class:
```

Comment:
one could imagine, that semantics of that are as follows:
instances which are defined while the frame class has this flag are updated on redefinition
instances which were defined when this flag was nil are not redefined no matter what

this is just a suggestion